update ingress docs

rayowang opened the pull request:
fixes: #4039

rayowang commented on site/content/docs/user/ingress.md:
```diff
@@ -52,15 +56,10 @@ nodes:
-nodeSelector:
-  kubernetes.io/hostname: "kind-control-plane"
-{{< /codeFromInline >}}
+If you want to run with multiple nodes, the built-in Ingress (since kind v0.9.0) will automatically use the configured PortMapping, no additional nodeSelector is required.
```
> this is not accurate, cloud-provider-kind does not apply to the `extraPortMapping` section ... @BenTheElder for thoughts, should we leave both "roll your ingress and do this network tricks" and cloud provider kind?

I've refined this a bit and will keep both scenarios for now. Clarify that Kind v0.9.0+ natively supports Ingress/Gateway API via its built-in cloud provider, eliminating the need to deploy third-party controllers by default; correct the multi-node cluster configuration description (the built-in Ingress does not rely on extraPortMapping, while third-party controllers may require nodeSelector configuration, which is not recommended for most users); and add information on version compatibility and Gateway API.

aojea commented on site/content/docs/user/ingress.md:
```diff
@@ -52,15 +56,16 @@ nodes:
-{{< /codeFromInline >}}
+  {{< /codeFromInline >}}
 
 ### Ingress NGINX
```
can we please remove this section now that the retirement is official ?

https://kubernetes.io/blog/2025/11/11/ingress-nginx-retirement/

aojea commented on site/content/docs/user/ingress.md:
```diff
@@ -8,25 +8,29 @@ menu:
 
-Create a kind cluster and run [Cloud Provider KIND]
-to enable the loadbalancer controller which ingress-nginx will use through the loadbalancer API.
+Create a kind cluster. The built-in cloud provider (cloud provider kind, enabled by default since v0.9.0) will automatically enable the LoadBalancer controller for Ingress to use.
```
i not a built-in cloud provider, you need to run it separately

aojea commented:
/lgtm
/approve

I think is good enough we can iterate

aojea commented on site/content/docs/user/ingress.md:
```diff
@@ -8,25 +8,29 @@ menu:
+- using a [LoadBalancer]. 
+- leverage KIND's `extraPortMapping` config option when creating a cluster to forward ports from the host.
+
+> **NOTE**: Since kind v0.9.0, the built-in cloud provider (cloud provider kind) natively supports Ingress. You do not need to deploy a third-party Ingress controller (e.g., Ingress NGINX, Contour) by default.
```
```suggestion
> **NOTE**: Since cloud-provider-kind v0.9.0, it natively supports Ingress. You do not need to deploy a third-party Ingress controller (e.g., Ingress NGINX, Contour) to use this functionality```

aojea commented on site/content/docs/user/ingress.md:
```diff
@@ -8,25 +8,29 @@ menu:
 
-Create a kind cluster and run [Cloud Provider KIND]
-to enable the loadbalancer controller which ingress-nginx will use through the loadbalancer API.
+Create a kind cluster. The built-in cloud provider (cloud provider kind, enabled by default since v0.9.0) will automatically enable the LoadBalancer controller for Ingress to use.
```
```suggestion
Create a kind cluster and run [Cloud Provider KIND] to enable the loadbalancer controller which ingress controllers will use through the loadbalancer API. Since version 0.9.0 [Cloud Provider KIND] also supports Ingress implementation```

aojea replied:
remove Option 1 and Option 2, we do have only one option now

rayowang commented:
> left some comments, please let me know if you can address them or I will take over keeping you as co author

I’ve accepted the suggested changes, removed the labels for option 1 and option 2, and also squashed the commits, but kept the original content. Could you take a look and see if it’s okay now?
